### ISSUE TITLE
data-field tooltip

### ISSUE THREAD

**Issue by serlogo53** [wenzhixin/bootstrap-table#1634]
hi,

is it possible dynamic cell tooltip?
for example

 
 th tooltip="db_column2" data-field="db_column1" data-sortable="true">tag name</th

**Comment by dabros** [comment]
@serlogo53 - i see a few possibilities

https://jsfiddle.net/dabros/0h53twrh/

My example is simple, and there is some limitation in accessing the TD while having all row data accessible, but you can work around that pretty easy if thats an issue.

**formatter**

I first thought of the formatter column option

http://issues.wenzhixin.net.cn/bootstrap-table/#options/format.html

But you really get at the TD parent from there, as (this) is wrong context, and working via index is just too awkward. But you could wrap the value in a tag, give that tag "width:100%;display:block" - i use that for links in any tables i make, TD padding wouldnt be tooltiped but it might work for you.


**js callback**

If you want to apply titles directly to TD, prob best just use bootstrap table callback/event like onPostBody. 

I include an example in my jsfiddle, but basically something like this is all you need:

```javascript
   .on('post-body.bs.table', function (){

        $("#eventsTable tbody > tr > td").each(function(){
			$(this).attr('title',$(this).text());
        });
    })
```

You can make the new value anything you want, even use ajax.

Also remember you can set the 'data-unique-id' value to a column name, and every TR then has that value for easy access

```html
<tr data-index="3" data-uniqueid="bbGrid">...</tr>
```

**cellStyle (to limit js callback)**

A similiar function in terms of param, but allows you to easily apply conditional css or classes onto the cell.

The closest example just uses rowStyle and column header classes: http://issues.wenzhixin.net.cn/bootstrap-table/#options/table-style.html

But it is similiar approach and i did find these: #599, #378, #221 from https://github.com/wenzhixin/bootstrap-table/issues?utf8=%E2%9C%93&q=is%3Aissue+cellstyle

What i would do in this case is apply class to all cells you want dynamic titles on, or even use rowStyle if selective rows.

**Comment by wenzhixin** [comment]
Have you solved your problem @serlogo53?

**Comment by serlogo53** [comment]
i will try. give me time. :) :+1:

**Comment by serlogo53** [comment]
no i couldn't solved my problem. i want tooltip when i mouse over the cell. and it will take information second data-field. because i want to hide second field. i will show it on tooltip on mouseover. abros writes couldn't help me or i couldn't do it. master wenzhixin.

**Comment by wenzhixin** [comment]
ref #1632

**Comment by dabros** [comment]
@serlogo53 - then, as i showed in my examples, use a formatter and just set the row[x] to whatever you wanted, id suggest X number of formatters with unique names, that all call same code passing in col name and row

As you didnt seem to follow what i meant, here is an update of that fiddle:

https://jsfiddle.net/dabros/0h53twrh/6/

Though again, as i said above (and in js comments in fiddle) I suggest not messing with margins too much, just deal with it not '100%' each way filling the TD, rather than unexpected behaivour on other browsers or resolutions. But i included basic styling too so you can see what i meant.

Btw - use the @ + username syntax if talking to someone, else we maybe get email but no github notify icon so slower response usually
